### DISCUSSION TITLE
Maturin develop with pyproject.toml [project.optional-depenencies]

### DISCUSSION THREAD

**Discussion by softstream-link** [PyO3/maturin#1941] [Q&A] [answered]
is there a command line argument to Maturin develop that will install optional dependencies specified in the pyproject.toml file

```toml
[project.optional-dependencies]
test = ["whatever>=1.1"]
```

Why is this relevant in my use case. I have a rust extension which needs a third party library for testing and so I would like to add an optional test dependency using standard pyproject.toml format. 

I noticed that `Maturin develop` command will not only recompile the rust extension and drop the .so into the correct folder of the python extension module but also will install any pip dependencies specified in the `project.dependencies` section of the `.toml` however because this library many only be required for testing or a user may choose a differs library during runtime how does one avoid adding it to the eventual wheel as a dependency and only leave it in the `test` profile of the `project.optional-depencies`  but not having to install it manually after running `maturin develop`

```
[project] 
dependencies = ["whatever>=1.1"]
```

**Comment by messense** [comment, marked as answer]
You can use `maturin develop --extras test`.

**Reply by softstream-link** [reply]
Thank you
